awx/main/tasks/jobs.py

```diff
@@ -1266,6 +1266,11 @@ def build_extra_vars_file(self, project_update, private_data_dir):
             # for raw archive, prevent error moving files between volumes
             extra_vars['ansible_remote_tmp'] = os.path.join(project_update.get_project_path(check_if_exists=False), '.ansible_awx', 'tmp')
 
+        # TODO: Is this the right way to do this? -relrod
+        if project_update.project.signature_validation_credential is not None:
+            pubkey = project_update.project.signature_validation_credential.get_input('gpg_public_key')
+            extra_vars['gpg_pubkey'] = pubkey
```

Comment:
I get where this question is coming from now.

How else could we have done this? The history is that jobs used to have many credential ForeignKey relationships. It had `network_credential`, `cloud_credentials`, and others. Then we consolidated those into a single `credentials` relationship. Importantly, this is on the _unified job_ model, so even projects / project updates should have this arbitrary credentials relationship. If we tracked this new credential in that existing many-to-many, as opposed to introducing a new ForeignKey field, then we could define the `extra_vars` injector in the credential type definition, and there would be no code to write for it whatsoever.

However, I get why that's hard, weird, and non-obvious. For projects, we don't expose the `credentials` many-to-many  relationship in the UI _or_ the API. Exposing it would obligate us to write validation rules in the views and a bunch of other stuff. The UI would have to write more custom logic to specially help the user out so that they pick the right credential type (I guess they already add some stuff for this implementation). But most importantly, this would break the pattern we have with `scm_credential`. We could migrate that field into the generic relationship too, but then that creates even more work. This is a pre-existing inconsistency in the app.

Without getting into all of that, this is probably the most direct way to do the injection.

Reply:
I will keep this and work to prevent the credential from being used on normal job. I'll remove the TODO.

Reply:
Looking into this more, I am not sure how this credential type could be used in a normal job. If I try to apply it to a job via the API, I get:
```
Cannot assign a Credential of kind `cryptography`
```
from `JobTemplateCredentialsList#is_valid_relation`.

Is there anything to do here @AlanCoding?